wallet/src/Cardano/Wallet/Kernel/Transactions.hs

```diff
@@ -173,19 +175,30 @@ pay activeWallet spendingPassword opts accountId payees = do
         case res of
              Left e      -> return (Left $ PaymentNewTransactionError e)
              Right (txAux, partialMeta, _utxo) -> do
-                 succeeded <- newPending activeWallet accountId txAux partialMeta
-                 case succeeded of
-                      Left e   -> do
-                          -- If the next retry would bring us to the
-                          -- end of our allowed retries, we fail with
-                          -- a proper error
-                          retriesLeft <- applyPolicy retryPolicy rs
-                          return . Left $ case retriesLeft of
-                               Nothing ->
-                                   PaymentSubmissionMaxAttemptsReached
-                               Just _  ->
-                                   PaymentNewPendingError e
-                      Right meta -> return $ Right (taTx $ txAux, meta)
+                 let sz = fromIntegral $ BL.length $ serialize txAux
+                 maxSz <- Node.getMaxTxSize (walletPassive activeWallet ^. walletNode)
+                 if sz > maxSz then
```

Comment:
@KtorZ, just to make sure, we don't need `sz >= maxSz` here (most probably not, but asking just to be safe :) )

Reply:
I am not sure how cardano-sl approaches maximum actually and whether it treats it as an inclusive or exclusive boundary :thinking: ...